qiskit/circuit/add_control.py

```diff
@@ -32,6 +32,10 @@ def add_control(operation, num_ctrl_qubits, label):
             num_qubits + 2*num_ctrl_qubits - 1.
     """
     import qiskit.extensions.standard as standard
+    if operation.phase:
```

Comment:
I would like to be able to remove this clause and have the phase case handled by the standard `control` function without converting to a unitary first.